homeassistant/components/pi_hole/binary_sensor.py

```diff
@@ -25,16 +30,18 @@ class PiHoleBinarySensor(PiHoleEntity, BinarySensorEntity):
     """Representation of a Pi-hole binary sensor."""
 
     @property
-    def name(self):
+    def name(self) -> str:
         """Return the name of the sensor."""
         return self._name
 
     @property
-    def unique_id(self):
+    def unique_id(self) -> str:
         """Return the unique id of the sensor."""
         return f"{self._server_unique_id}/Status"
 
     @property
-    def is_on(self):
+    def is_on(self) -> bool:
         """Return if the service is on."""
-        return self.api.data.get("status") == "enabled"
+        if self.api.data.get("status") == "enabled":
+            return True
+        return False
```

Comment:
Another option:
```suggestion
        api_data: dict[str, str] = self.api.data
        return api_data.get("status") == "enabled"
```

Reply:
I do not agree with this suggestion. I think that this is shadowing the actual typing. If `data` needs typing, please that it at the source. Not randomly in the middle of the code.

Reply:
Thanks for the comment here @frenck, just opened a PR to partially type the pihole package.